app/logic/solana_logic.rb

```diff
@@ -495,8 +497,12 @@ def cli_request(cli_method, rpc_urls)
       end
 
       # byebug
-
-      return JSON.parse(response_utf8) unless response_utf8 == ''
+      if response_utf8 != ''
+        # Include errors from the solana-cli 
+        return JSON.parse(response_utf8).merge!('cli_error' => error_msg)
```

Comment:
I think you could refactor the conditions in your `if/elseif` statement. It's usually a code smell to see both statements with a `!` in them. I would recommend using `#present?` and `#blank?`